Add profile template and prompt for extending Atrium

mdeltito opened the pull request:
This generates a bare-minimum installation profile for use when extending Atrium. The prompt is currently only invoked when generating an Atrium project.

There is likely some discussion to be had around what constitutes a "bare-minimum" Atrium installation, but this set of default dependencies has been a sufficient and stable starting point.

Commit:
Add profile template and prompt for extending Atrium

grayside commented on lib/util.js:
```diff
@@ -8,6 +8,8 @@ module.exports.labelMaker = function(str) {
 module.exports.tokens = function(options) {
   return {
+    name: options.projectName,
+    nameFunc: options.projectName.replace(/-/, '_'),
```
For consistency across generator efforts, we should probably keep this as projectName and machineName. Also `replace()` for a simple case like this will accept a string.

mdeltito replied:
Good call, token names updated

grayside commented:
Nice! I'd been thinking something like this would be a powerful addition to Gadget.

Broad thought: Is Yeoman/Gadget the right spot to take on Drupal code generation? I think it's a fine first step, but especially as we move into D8 we'll want to either keep this to a minimum or thinking about leveraging Console.

Paging @mike-potter for review of "minimal Atrium".

Commit:
Update token names for consistency

mike-potter commented:
Yeah, we probably need to talk about what we mean by "minimal Atrium".  No matter what we do, people will be confused by this terminology.  From a technical perspective, I think this looks reasonable as the "minimum set of Atrium dependencies".  But we probably need a better title for it since "Atrium" implies a lot of other functionality.

Also, much like we never use "minimal Drupal" I'm a bit concerned about where we would use this vs a "recommended Atrium" profile.  For example, it doesn't hurt to have stuff like oa_related, oa_comment, oa_files, oa_discussion, etc enabled on a site if you just don't use them.  But then they are there to easily configure as needed.  If it doesn't hurt a site performance, shouldn't we have a more common set of "base Atrium" that has the functionality that people are expecting?

Honestly, what I really _LIKE_ to see rather than a "minimal" profile is to expand this by adding all of the other modules that are enabled in Atrium by default and then comment them out with comments indicating what functionality is being commented out.  For example:

; Provides Events and Calendars
; oa_events

; Provides Discussion Forums
; oa_discussion

; Adds several paragraph styles to the Body field for Related content and Media
; oa_related

; Adds the ability to enable clean Comments on content types
; oa_comment

...etc.  Then it's a lot easier for a project to un-comment the functionality they need

mirie committed:
Merge branch 'refs/heads/master' into feature/oa-profile-extend

mirie committed:
Updated Open Atrium custom profile option.

- Updated Atrium profile template files
- Added commented out optional apps in the profile.info template
- Updated the paths for OA template files
- Made custom profile true by default

mirie commented:
@mike-potter, @grayside, @mdeltito: I updated the profile templates and adjusted the path location for the templates.

mirie commented:
I'm not sure why there is this Travis error:

```
1) gadget:app "before all" hook:
     Uncaught TypeError: Cannot read property 'id' of undefined
      at Object.prompts.when (lib/prompts.js:68:37)
```

Looks like it's complaining about the following lines:

```
{
    type: 'confirm',
    name: 'drupalDistroCreateProfile',
    message: function(answers) {
      return 'Would you like to extend '
        + chalk.red(answers['drupalDistro'].option.name)
        + ' with a custom profile?';
    },
    when: function(answers) {
      return answers['drupalDistro'].id == 'openatrium';  // <<< This line
    },
    default: function(answers) {
      return false;
    }
  }
```

mdeltito commented:
@mike-potter in terms of "minimal Atrium" my intention was only to define what the scaffolding for an extended profile would look like. I agree that the terminology here is important so that it is clear this is a starting point for your own code, and NOT a recommended minimal starting point for functionality. 

The suggestion to add commented lines to the profile.info makes a lot of sense, and it looks like @mirie took care of that in the latest commits. Nice!

arithmetric commented:
@grayside Assigning this to you for review.

grayside commented:
This has been floating a long time, and since we do not see many atrium projects started with the generator these days, I'm thinking this should be revisited more generally in how we think about profile setup.
